### ISSUE TITLE
tf.reduce_sum duplicate-axis validation differs between eager/XLA and autoclustering

### ISSUE THREAD

**Issue by MRiffiAslett** [tensorflow/tensorflow#119360]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

No

### Source

source

### TensorFlow version

2.20.0-dev0+selfbuilt

### Custom code

Yes

### OS platform and distribution

Linux-6.14.0-37-generic-x86_64-with-glibc2.39

### Mobile device

N/A

### Python version

3.12.3

### Bazel version

bazel 9.1.0, invoked via Bazelisk

### GCC/compiler version

gcc (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0

### CUDA/cuDNN version

No CUDA/cuDNN. CPU-only environment.

### GPU model and memory

No CUDA/cuDNN. CPU-only environment.

### Current behavior?

`tf.reduce_sum` behaves differently depending on how XLA is enabled when duplicate axes are passed.

Eager execution rejects the input. Explicit XLA compilation with `jit_compile=True` also rejects the input. However, the autoclustering path accepts the input and returns a value.

This looks like an inconsistent validation issue rather than a numerical correctness issue on valid input, since the TensorFlow documentation says that entries in `axis` must be unique.

Expected behavior: autoclustering should preserve the same user-visible TensorFlow semantics as eager execution. In this case, duplicate axes should be rejected rather than silently accepted.

### Standalone code to reproduce the issue

```shell
import os

# Must be set before importing TensorFlow.
os.environ["TF_XLA_FLAGS"] = "--tf_xla_auto_jit=2"

import tensorflow as tf

tf.config.optimizer.set_jit(True)

x = tf.constant([[[[[[3.5]]]]]])
axis = [-2, -1, -2, -1, -2, -1]

print("TensorFlow:", tf.__version__)

try:
    print("eager      :", tf.reduce_sum(x, axis).numpy())
except Exception as e:
    print("eager      :", type(e).__name__, str(e).splitlines()[0])

@tf.function(jit_compile=True)
def f_xla(t):
    return tf.reduce_sum(t, axis)

try:
    print("xla        :", f_xla(x).numpy())
except Exception as e:
    print("xla        :", type(e).__name__, str(e).splitlines()[0])

@tf.function
def f_ac(t):
    return tf.reduce_sum(t, axis)

try:
    print("autocluster:", f_ac(x).numpy())
except Exception as e:
    print("autocluster:", type(e).__name__, str(e).splitlines()[0])
```

### Relevant log output

```shell
TensorFlow: 2.20.0-dev0+selfbuilt
eager      : InvalidArgumentError Invalid reduction arguments: Axes contains duplicate dimension: 4
xla        : InvalidArgumentError Invalid reduction arguments: Axes contains duplicate dimension: 4
autocluster: [[[[3.5]]]]
```
